packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -101,13 +101,30 @@ const double _kMinButtonSize = kMinInteractiveDimension;
 /// The default [IconButton] is the standard type, and contained icon buttons can be produced
 /// by configuring the [IconButton] widget's properties.
 ///
+/// Material Design 3 also treats [IconButton]s as toggle buttons. In order
+/// to not break existing apps, the toggle feature can be optionally controlled
+/// by the [isSelected] property.
+///
+/// If [isSelected] is null it will behave as a normal button. If [isSelected] is not
+/// null then it will behave as a toggle button. If [isSelected] is true then it will
+/// show [selectedIcon], if it false it will show the normal [icon].
+/// Pressing the button will toggle the state.
```

Comment:
I just realized that that isn't automatic, is it? They have to provide an onPressed handler that changes the state and passes it into the constructor. We should take that line out then.

Reply:
Thanks a lot for the suggestion. Yes. We need to change the state using onPressed. Fixing!